overlord/devicestate/devicestate.go

```diff
@@ -521,7 +521,7 @@ func remodelEssentialSnapTasks(ctx context.Context, st *state.State, pathSI *pat
 		// which needs to be installed
 		return remodelVar.InstallWithDeviceContext(ctx, st,
 			pathSI, ms.newSnap, newModelSnapChannel, userID,
-			snapstate.Flags{}, deviceCtx, fromChange)
+			snapstate.Flags{Required: true}, deviceCtx, fromChange)
```

Comment:
An unrelated change, but I believe this should be set to true, no?

Reply:
This should ideally be its own PR, and it would be nice to add a check for this to be set in either an existing unit test or a new one

Reply:
After asking some more questions, this change isn't needed, as we prevent essential snaps from being removed by other means.